app/middleware/getServices.js

```diff
@@ -0,0 +1,75 @@
+const VError = require('verror').VError;
+const constants = require('../lib/constants');
+const elasticsearchClient = require('../lib/elasticsearchClient');
+const esGeoQueryBuilder = require('../lib/esGeoQueryBuilder');
+const esGetServiceHistogram = require('../lib/promHistograms').esGetServices;
+const esQueryLabelName = require('../lib/constants').promEsQueryLabelName;
+const serviceDataMapper = require('../lib/utils/serviceDataMapper');
+const log = require('../lib/logger');
+
+function handleError(error, next) {
+  const errMsg = 'Error with ES';
+
+  log.error({ err: new VError(error.stack, errMsg) }, errMsg);
+  next(error);
+}
+
+function mapResults(results, res) {
+  res.locals.services = results.hits.hits.map((result) => {
+    // eslint-disable-next-line no-underscore-dangle
+    const service = result._source;
+
+    if (service) {
+      if (result.sort) {
+        service.distance = result.sort[0];
+      }
+
+      if ((service.address) && (service.address.addressLines) && (service.address.postcode)) {
```

Comment:
Suggest this if is removed from this function and pushed into `addressFormatter`. There could be some unit tests added for the `serviceDataMapper` too.